HW1/P2/P2b.py

```diff
@@ -1,3 +1,34 @@
 from P2 import *
 
-# Your code here
+# Initialize Spark
+from pyspark import SparkContext
+sc = SparkContext()
+sc.setLogLevel("ERROR")
+
+# Define key parameters
+coordWidth = 2000
+coordHeight = 2000
+noPartitions = 100
+
+# Create RDD with i, j coordinates
+pixCoordIJ = sc.parallelize([(x, y) for x in range(coordWidth) for y in range(coordHeight)])
+
+# Apply the Mandelbrot function & set number of partitions
+pixMandel = pixCoordIJ.map(lambda (i, j): ((i, j), mandelbrot(j/500.0 - 2, i/500.0 - 2))).partitionBy(noPartitions)
```

Comment:
Also, when you rerun the program several times, did your graphs change or were they pretty consistent? I'm also trying to understand why partitionBy seems to have worked.